test_vae_training/2_check_trained_model.py

```diff
@@ -0,0 +1,214 @@
+
+model_decoder_file = glob.glob(os.path.join(NN_dir, "*_decoder_model.h5"))[0]
+
+weights_decoder_file = glob.glob(os.path.join(NN_dir, "*_decoder_weights.h5"))[0]
+
+# Load saved models
+loaded_model = load_model(model_encoder_file)
+loaded_decode_model = load_model(model_decoder_file)
+
+loaded_model.load_weights(weights_encoder_file)
+loaded_decode_model.load_weights(weights_decoder_file)
+
+# +
+# Encode normalized compendium into latent space
+compendium_encoded = loaded_model.predict_on_batch(normalized_compendium)
+
+compendium_encoded_df = pd.DataFrame(
+    data=compendium_encoded, index=normalized_compendium.index
+)
+
+# +
+# Get and save PCA model
+model = pca.fit(compendium_encoded_df)
+# model = umap.UMAP(random_state=random_state).fit(compendium_encoded_df)
+
+latent_compendium_PCAencoded = model.transform(compendium_encoded_df)
+
+latent_compendium_PCAencoded_df = pd.DataFrame(
+    data=latent_compendium_PCAencoded,
+    index=compendium_encoded_df.index,
+    columns=["1", "2"],
+)
+
+# +
+# Plot umap of encoded data
+fig = ggplot(latent_compendium_PCAencoded_df, aes(x="1", y="2"))
+fig += geom_point(alpha=0.2)
+fig += labs(x="PCA 1", y="PCA 2", title="PCA encoded normalized compendium")
+fig += theme_bw()
+fig += theme(
+    legend_title_align="center",
+    plot_background=element_rect(fill="white"),
+    legend_key=element_rect(fill="white", colour="white"),
+    legend_title=element_text(family="sans-serif", size=15),
+    legend_text=element_text(family="sans-serif", size=12),
+    plot_title=element_text(family="sans-serif", size=15),
+    axis_text=element_text(family="sans-serif", size=12),
+    axis_title=element_text(family="sans-serif", size=15),
+)
+fig += guides(colour=guide_legend(override_aes={"alpha": 1}))
+
+print(fig)
+# -
+
+# ## Simulate vs real data
+#
+# Here we want to make sure that the distribution of the simulated data looks similar to the real data -- i.e. the sparsity is similar.
+
+simulated_data = simulate_expression_data.run_sample_simulation(
+    loaded_model, loaded_decode_model, normalized_compendium, 5
+)
+
+simulated_data.head()
+
+# How sparse is this data matrix?
+# How many 0s across the entire matrix?
+num_zeros = (simulated_data == 0).sum().sum()
+total = simulated_data.shape[0] * simulated_data.shape[1]
+num_zeros / total
+
+# How lowly expressed is this data matrix?
+num_low = (simulated_data < 0.1).sum().sum()
+num_low / total
+
+# **Takeaways:**
+# * Latent representation captures the distribution of the input dataset, which is mainly composed of a single large cluster.
+# * The latent space compresses the spreading seen in the input dataset, as previously seen
+# * The simulated data, generated by randomly sampling from the latent space, is generating all lowly expressed microbes (i.e. all expression is below 0.1, but none are 0), which makes sense given the dominant signal in the data are these 0 counts
```

Comment:
This is interesting! Do you think the VAE is learning each gene's mean (which are skewed down by the zeros), then simulating data around it?

Reply:
Just putting what you said offline here for reference:

```
Oh I was thinking that to minimize reconstruction error you'd probably want to simulate data with a mean centered around the Gene's mean
And since the gene's mean is skewed down by the zeros lots of your samples have low expression value
``

So yes, that is what I was thinking too. Since most genes have 0 counts, then the latent variables distribution will be skewed to 0 so when the simulation goes to sample from the latent space that is why the values are so low. I'll make sure I'll update my comment here to be more clear